src/Proxy.NET/Authentication/JwtAuthenticationHandler.cs

```diff
@@ -0,0 +1,35 @@
+using System.Security.Claims;
+using System.Text.Encodings.Web;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.Extensions.Options;
+using Proxy.NET.Services;
+
+namespace Proxy.NET.Authentication;
+
+public class JwtAuthenticationHandler(
+    IOptionsMonitor<ProxyAuthenticationOptions> options,
+    IAuthorizeService authorizeService,
+    ILoggerFactory logger,
+    UrlEncoder encoder
+) : AuthenticationHandler<ProxyAuthenticationOptions>(options, logger, encoder)
+{
+    protected override async Task<AuthenticateResult> HandleAuthenticateAsync()
+    {
+        if (!Request.Headers.TryGetValue("x-ms-client-principal", out var jwt) || string.IsNullOrEmpty(jwt))
+            return AuthenticateResult.Fail("Authentication failed.");
+
+        var requestContext = authorizeService.GetRequestContextFromJwt(jwt!);
+        if (requestContext is null)
+            return AuthenticateResult.Fail("Authentication failed.");
+
+        Context.Items["RequestContext"] = requestContext;
+
+        var identity = new ClaimsIdentity(null, nameof(ApiKeyAuthenticationHandler));
+        var ticket = new AuthenticationTicket(new ClaimsPrincipal(identity), Scheme.Name);
+
+        // Awaiting a completed task to suppress the warning
+        await Task.CompletedTask;
```

Comment:
This is not really a good idea, there's two different ways that we can tackle it.

1. Remove the `async` from the method signature and have the return statements be `Task.FromResult(...)`, which wraps a return value in a completed task.
2. Change the signature of `GetRequestContextFromJwt` to be an async method, even if it's not doing anything async, and have it return `Task.FromResult(jwt)`.

I'd be more in favour of the second, as it leaves room for future where we may need to do something in the service call with the JWT.